.github/scripts/compile.sh

```diff
@@ -22,15 +22,41 @@
 # https://sipb.mit.edu/doc/safe-shell
 set -Eefuxo pipefail
 
-WORKING="$(pwd)"
 BUILD_FLAGS="$(rpmbuild --eval '%set_build_flags')"
 eval "${BUILD_FLAGS}"
 
-#region libwebsockets
-wget "${LWS_SOURCE_URL}" -O libwebsockets.tar.gz
-tar -zxf libwebsockets.tar.gz --one-top-level=lws-src --strip-components 1
 
-cmake -S "$WORKING/lws-src" -B "$WORKING/lws_build" \
+if [ -z "${REMOTE_SOURCES_DIR:-}" ]; then
+  # If no REMOTE_SOURCES_DIR present in env, we use $(pwd) as working dir
+  WORKING_DIR=$(pwd)
+  SKUPPER_DIR=$WORKING_DIR
+else
+  # If the env contains REMOTE_SOURCES_DIR, we will use that as the working dir
+  # If REMOTE_SOURCES_DIR is provided, this scripts expects the following -
+  # 1. proton sources to be in $REMOTE_SOURCES_DIR/app/proton
+  # 2. libwebsockets sources to be in $REMOTE_SOURCES_DIR/app/libwebsockets
+  # 3. skupper-router sources to be in $REMOTE_SOURCES_DIR/app/skupper-router
+  WORKING_DIR=$REMOTE_SOURCES_DIR
+  SKUPPER_DIR=$WORKING_DIR/skupper-router/app
+  cd $WORKING_DIR
+fi
+
+#region libwebsockets
+LWS_DIR=$WORKING_DIR/libwebsockets/app
+LWS_BUILD_DIR=$LWS_DIR/build
+LWS_INSTALL_DIR=$LWS_DIR/install
+if [ -z "${REMOTE_SOURCES_DIR:-}" ]; then
+  # $REMOTE_SOURCES_DIR was not provided, we will have to download the libwebsockets source from ${LWS_SOURCE_URL}
+  # Get the libwebsockets source into a libwebsockets.tar.gz file
+  # and untar it into the libwebsockets folder
+  wget ${LWS_SOURCE_URL} -O libwebsockets.tar.gz
+  tar -zxf libwebsockets.tar.gz --one-top-level=$LWS_DIR --strip-components 1
+fi
```

Comment:
It might make sense to move all downloading and setting of these location-related variables to the top of the file. And do this in a single place for both lws and proton. So that there is only one `if` that branches on whether we are using the prepared `/app` source dirs in network-restricted build, or whether we are downloading things with wget